qa/tasks/cephfs/test_volumes.py

```diff
@@ -1247,7 +1247,7 @@ def test_subvolume_group_quota_exceeded_subvolume_removal_retained_snaps(self):
             # For quota to be enforced
             time.sleep(20)
             # create 400 files of 1MB to exceed quota
-            self._do_subvolume_io(subvolname, subvolume_group=group, number_of_files=400)
+            self._do_subvolume_io(subvolname, subvolume_group=group, create_dir='dir2', number_of_files=400)
```

Comment:
If there are more than one cap flush requests come in a short time and if the have same parents, then the MDS will only update the parents' `rstat` once in `mds_dirstat_min_interval` seconds to avoid updating too often. That means the `rstat` updating will be deferred when the next `tick`, which is `5s`, is fired.

Only when the parents' `rstat` are updated will the quota info could be broadcast to the corresponding clients. 

And also the client could cache the dirty caps up to `(60+1)` seconds in `kclient` and (5+1) seconds in `libcephfs`.

That means the client in worst case will see the quota change after `(61+5)` seconds.

Reply:
> If there are more than one cap flush requests come in a short time and if the have same parents, then the MDS will only update the parents' `rstat` once in `mds_dirstat_min_interval` seconds to avoid updating too often. That means the `rstat` updating will be deferred when the next `tick`, which is `5s`, is fired.

This check is for the immediate parent right (as far as I can see from predirty_journal_parents())?

> 
> Only when the parents' `rstat` are updated will the quota info could be broadcast to the corresponding clients.
> 
> And also the client could cache the dirty caps up to `(60+1)` seconds in `kclient` and (5+1) seconds in `libcephfs`.

Umm.. I wonder why there is a timeout difference...

> 
> That means the client in worst case will see the quota change after `(61+5)` seconds.

ACK.

Reply:
> > If there are more than one cap flush requests come in a short time and if the have same parents, then the MDS will only update the parents' `rstat` once in `mds_dirstat_min_interval` seconds to avoid updating too often. That means the `rstat` updating will be deferred when the next `tick`, which is `5s`, is fired.
> 
> This check is for the immediate parent right (as far as I can see from predirty_journal_parents())?
> 

Correct.

> > Only when the parents' `rstat` are updated will the quota info could be broadcast to the corresponding clients.
> > And also the client could cache the dirty caps up to `(60+1)` seconds in `kclient` and (5+1) seconds in `libcephfs`.
> 
> Umm.. I wonder why there is a timeout difference...
> 

The libcephfs just set it to `5`, while kclient will be a little complex and will set it a range `[min, max] == [5, 60]`.

> > That means the client in worst case will see the quota change after `(61+5)` seconds.
> 
> ACK.